client-app/ui-kit/composables/useSmartSticky.ts

```diff
@@ -0,0 +1,423 @@
+    const element = toValue(stickyElement);
+
+    if (!element || !toValue(container)) {
+      return;
+    }
+
+    dimensions.value.viewportHeight = globalThis.window.innerHeight;
+    dimensions.value.viewportTop = getScrollPosition();
+
+    dimensions.value.containerTop = containerBounding.top.value + dimensions.value.viewportTop;
+    dimensions.value.containerHeight = containerBounding.height.value;
+
+    dimensions.value.elementHeight = elementBounding.height.value;
+    dimensions.value.elementWidth = elementBounding.width.value;
+
+    dimensions.value.topSpacing = topSpacingResolved.value;
+    dimensions.value.bottomSpacing = bottomSpacingResolved.value;
+  }
+
+  function calculatePosition() {
+    if (!isEnabled.value) {
+      resetPosition();
+      return;
+    }
+
+    const element = toValue(stickyElement);
+    const containerEl = toValue(container);
+
+    if (!element || !containerEl) {
+      return;
+    }
+
+    updateDimensions();
+
+    const dims = dimensions.value;
+    const { containerTop, containerHeight, elementHeight, viewportHeight, viewportTop, topSpacing, bottomSpacing } =
+      dims;
+
+    const containerBottom = containerTop + containerHeight;
+    const elementTop = elementBounding.top.value + viewportTop;
+
+    const isElementTallerThanViewport = elementHeight + topSpacing + bottomSpacing > viewportHeight;
+
+    const isScrollingDown = viewportTop > dims.lastViewportTop;
+    const isScrollingUp = viewportTop < dims.lastViewportTop;
+
+    const affixType = getAffixType({
+      isTaller: isElementTallerThanViewport,
+      isScrollingDown,
+      isScrollingUp,
+      containerTop,
+      containerBottom,
+      viewportTop,
+      viewportHeight,
+      elementHeight,
+      elementTop,
+      topSpacing,
+      bottomSpacing,
+    });
+
+    applyStyles(affixType, dims, elementTop, containerTop);
+
+    isStuckTop.value = affixType === AFFIX_TYPES.FIXED_TOP;
+    isStuckBottom.value = affixType === AFFIX_TYPES.FIXED_BOTTOM;
+    isActive.value = affixType !== AFFIX_TYPES.STATIC;
+
+    dims.lastViewportTop = viewportTop;
+  }
+
+  function applyStyles(affixType: AffixType, dims: IDimensions, elementTop: number, containerTop: number) {
+    const element = toValue(stickyElement);
+
+    if (!element) {
+      return;
+    }
+
+    const baseStyle: CSSProperties = {
+      left: "auto",
+      right: "auto",
+      width: `${dims.elementWidth}px`,
+    };
+
+    const stickyTopStyle: CSSProperties = {
+      ...baseStyle,
+      position: "sticky",
+      top: `${dims.topSpacing}px`,
+      bottom: "auto",
+    };
+
+    const stickyBottomStyle: CSSProperties = {
+      ...baseStyle,
+      position: "sticky",
+      top: "auto",
+      bottom: `${dims.bottomSpacing}px`,
+      alignSelf: "flex-end",
+    };
+
+    const absoluteStyle: CSSProperties = {
+      ...baseStyle,
+      position: "absolute",
+      top: `${translate.value}px`,
+      bottom: "auto",
+    };
```

Comment:
Perhaps we can replace it with a builder function just to avoid duplication, like this
```
function buildStyle(style: CSSProperties) {
  return {
    left: "auto",
    right: "auto",
    ...style,
  };
}

const stickyTopStyle = buildStyle({
  position: "sticky",
  top: `${dims.topSpacing}px`,
  bottom: "auto",
  width: `${dims.elementWidth}px`,
});
```